### ISSUE TITLE
[BUG] dbx v0.6+ / dbx deploy doesn't take the expected wheel file

### ISSUE THREAD

**Issue by copdips** [databrickslabs/dbx#321]
## Expected Behavior

In my `setup.py`, my project package name is `pr-cicd_test`, so I expect `dbx deploy` will pick up the wheel file with name `pr-cicd_test-xxx.whl`, and inject it in the final databricks json file as library.

## Current Behavior

Only for dbx v0.6+, it varies on which wheel file is built in the last in the `dist/` folder, which unfortunately is not `pr-cicd_test-xxx.whl`.

## Steps to Reproduce (for bugs)

Add for example `pyspark==3.2.0` in `requirements.txt` file, and run `dbx deploy` in a clean environment, for example in CICD without any wheel cache.
As pyspark needs to be built on the fly (see: the line `Preparing metadata (setup.py)` in the log), I don't know how pip wheel parses and sorts the requirements.txt file, but pyspark is considered as last built wheel file in the `dist/` folder:
https://github.com/databrickslabs/dbx/blob/4ed038c353ddc252be4acabfef7ed82e40d8fb76/dbx/utils/common.py#L63-L70

```log
...
Collecting pyspark==3.2.0
  Downloading https://xxxxx/_packaging/daa86ee5-06b8-417b-bc88-e64e3e2eef29/pypi/download/pyspark/3.2/pyspark-3.2.0.tar.gz (281.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 281.3/281.3 MB 9.5 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
...
Building wheels for collected packages: pr-cicd-test, pyspark
  Building wheel for pr-cicd-test (setup.py) ... done
  Created wheel for pr-cicd-test: filename=pr_cicd_test-0.1-py3-none-any.whl size=5499 sha256=3bdfb50c2b121fd257d9932be1e4121f0f1b64ab9abb24829607bc0cc28b3c62
  Stored in directory: /tmp/pip-ephem-wheel-cache-p1v0wtpg/wheels/a6/8d/77/a1ddc5ad8c61f36fa0ff1c6d882f0bef098007c2ebd98e0090  
  Building wheel for pyspark (setup.py) ... done
  Created wheel for pyspark: filename=pyspark-3.2.0-py2.py3-none-any.whl size=281805911 sha256=15945ef8d1b6758af56ec7593f5f56002232cd41b124b414076760a523001aef
  Stored in directory: /tmp/pip-ephem-wheel-cache-p1v0wtpg/wheels/44/6c/64/be399d14a9a08ef636c35df03c6a7a69fce562966742389792
Successfully built pr-cicd-test pyspark
```
Another method to repro is to run `pip wheel -w dist -e . --prefer-binary --no-cache-dir` to ensure no cache is used.

My current workaround is to run manually `pip wheel -w dist -e . --prefer-binary` right before `dbx deploy` which will cache all the dependencies' wheel files, then during the `dbx deploy` time, the internal `pip wheel -w dist -e . --prefer-binary` will find the `pyspark` wheel file in the cache (see the line `Using cached pyspark-3.2.0-py2.py3-none-any.whl` in the below logs), so won't built it, and as a result, only `pr-cicd_test` is rebuilt, so it's the last one too, which is good to us.

```log
...
Collecting pyspark==3.2.0
  Using cached pyspark-3.2.0-py2.py3-none-any.whl
  File was already downloaded /home/xiang/git/cicd-test/dist/pyspark-3.2.0-py2.py3-none-any.whl
...
Building wheels for collected packages: pr-cicd-test
  Building wheel for pr-cicd-test (setup.py) ... done
  Created wheel for pr-cicd-test: filename=pr_cicd_test-0.1-py3-none-any.whl size=5499 sha256=4cdac23ae74378b2d552d9d537166940b9c5ce10b61d789697f1c96004c46a95
  Stored in directory: /tmp/pip-ephem-wheel-cache-p3mzoe21/wheels/a6/8d/77/a1ddc5ad8c61f36fa0ff1c6d882f0bef098007c2ebd98e0090
Successfully built pr-cicd-test
```

## Your Environment

* dbx version used: dbx v0.6+
* Databricks Runtime version:  10.4

**Comment by copdips** [comment]
@renardeinside 
Just would like to know if you've seen this issue.

**Comment by renardeinside** [comment]
hi @copdips , 
this error means that there are multiple .whl files in the dist folder. I assume reason is that `pip wheel` command. I'm currently working on a resolution to make sure we won't run into this issue again.

**Comment by renardeinside** [comment]
Addressed in the relevant PR, will be released in 0.6.5 
Thanks a lot for pointing this out @copdips !
